prefect_sqlalchemy/credentials.py

```diff
@@ -0,0 +1,98 @@
+"""Credential classes used to perform authenticated interactions with SQLAlchemy"""
+
+from dataclasses import dataclass
+from typing import TYPE_CHECKING, Any, Dict, Optional
+
+from sqlalchemy.engine.url import URL
+from sqlalchemy.ext.asyncio import create_async_engine
+
+if TYPE_CHECKING:
+    from sqlalchemy.ext.asyncio.engine import AsyncConnection
+
+
+@dataclass
+class SQLAlchemyCredentials:
+    """
+    Dataclass used to manage authentication with SQLAlchemy.
+    The engine should be disposed manually at the end of the flow.
+
+    Args:
+        driver: The driver name, e.g. "postgresql+asyncpg"
+        database: The name of the database to use.
+        user: The user name used to authenticate.
+        password: The password used to authenticate.
+        host: The host address of the database.
+        port: The port to connect to the database.
+        query: A dictionary of string keys to string values to be passed to
+            the dialect and/or the DBAPI upon connect. To specify non-string
+            parameters to a Python DBAPI directly, use connect_args.
+        connect_args: The options which will be passed
+            directly to the DBAPI's connect() method as
+            additional keyword arguments.
+    """
+
+    driver: str
```

Comment:
I'm tempted to make `driver` an Enum so that we only receive drivers that we know we can support. What are your thoughts on that?